strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/event/artifact/MavenArtifactDeletedEventListener.java

```diff
@@ -43,11 +44,12 @@ public void handle(final ArtifactEvent event)
                                                                           event.getRepositoryId(),
                                                                           event.getPath());
 
-
-            mavenIndexGroupRepositoryComponent.cleanupGroupsContaining(event.getStorageId(),
-                                                                       event.getRepositoryId(),
-                                                                       event.getPath());
-
+            if (maybeMavenIndexGroupRepositoryComponent.isPresent())
```

Comment:
@fuss86 Sorry for disturbing, but don't you think that this is "tight coupling" here to check this optional every time?

Reply:
Maven indexer was already tight coupled with the maven layout provider. This PR intention is to make maven indexer optional,  based on the value of the single property.

Reply:
it's not only in MavenLayout provider, I mean that tight copling is that classes which inject Optional dependencies will depend on these dependencies internal configuration and need to check this optional is present.

Reply:
ok, I'll try to refactor it a little bit :) Thanks :)